app/scripts/modules/core/src/cluster/ClusterPostSearchResultSearcher.ts

```diff
@@ -42,14 +49,54 @@ export class ClusterPostSearchResultSearcher implements IPostSearchResultSearche
       }
     }), 'cluster');
 
-    const formatter: ISearchResultFormatter = searchResultFormatterRegistry.get(ClusterPostSearchResultSearcher.TYPE);
-    return this.$q.when([{
-      id: ClusterPostSearchResultSearcher.TYPE,
-      category: ClusterPostSearchResultSearcher.TYPE,
-      icon: formatter.icon,
-      iconClass: '',
-      order: formatter.order,
-      results: clusters
-    }]);
+    // for performance reasons, request each server group separately rather than all together and flag them as to
+    // whether or not they exist.
+    const requests: IPromise<IServerGroup>[] =
+      inputs.map((serverGroup: IServerGroupSearchResult) => {
```

Comment:
This seems like a lot of traffic for bigger result sets (e.g. `api`). If I'm reading this correctly, a result set of 400 server groups would result in 400 fetches all at once (limited only by the browser's concurrent request limits).

Feels like this would get throttled pretty quickly. Is there no way to do this validation on the backend? Seems like it would be a lot more efficient than bouncing each call through the network stack.

Reply:
Yea, it is a lot - this was the recommended solution from Cam.  Do you mean like sending the full list of ASGs up to, say, Gate, and having it query every one and then sending back?  I could perf test that.

Reply:
Embarrassingly, I had a dream about this and have an idea.  Will try this weekend.

Reply:
@anotherchrisberry added https://github.com/spinnaker/clouddriver/pull/1951 to cull bad ASGs during initial search, https://github.com/spinnaker/clouddriver/pull/1952 to allow bulk ASG retrieval via comma-delimited string of basic, pipe-delimited ASG info (for faster retrieval), and https://github.com/spinnaker/gate/pull/462 to expose in gate.  I'm gonna want cam, robz, or AJ to also take a look and make sure what I did passes muster performance-wise.  I left some comments about naming and the use of a POST (which I don't love), so if you have any thoughts there, I'd love to get them as well.